src/storage/src/client.rs

```diff
@@ -313,6 +277,200 @@ pub(crate) mod info {
     }
 }
 
+pub struct ReadObject<'a> {
+    client: &'a Storage,
+    request: control::model::ReadObjectRequest,
```

Comment:
In a future PR: we may want a way to make `Storage` mockable, and doing so may require introducing some kind of stub at this layer.